### PR TITLE
Test on real data

### LINKED ISSUE #24: Add tests on real datasets
When real data become available in the package as part of #2 , we should add tests that run on it.

### PR DESCRIPTION
**Pull request by fnattino**
Closes #24 

I have also resized the packaged data, limiting the street network to the city boundary. The main reason is that in the outskirt of Bucharest I have found one motorway link that would give rise to a self-intersecting-yet-perfectly-valid stroke, but which would then be tricky to validate as implemented in the test (i.e. using `sf::st_contains` would not identify the constituting initial edges due to the self-intersection). Since this is some sort of an edge case which would only make the validation trickier without bringing more insight into the well-functioning of rcoins, I have decided for the easy fix i.e. to modify the dataset. As a derived benefit, the data script is also more concise (the "buffer" functionality was imported from CRiSp but not really needed).

### CODE REVIEW

File: data-raw/bucharest.R [hunk 1]
```diff
@@ -51,24 +51,16 @@ get_osm_river <- function(river_name, bb, crs) {
   river_centerline <- osmdata_as_sf("waterway", "river", bb)
   river_centerline <- river_centerline$osm_multilines |>
     filter(.data$name == river_name) |>
+    st_filter(sf::st_as_sfc(bb), .predicate = sf::st_intersects) |>
```
Comment (fnattino) [inline data-raw/bucharest.R:54]: As also noticed in CRiSp, the Overpass API query for the river centerline now returns an additional branch of the Dambovita, not sure why, since it does not intersects the bounding box...

File: tests/testthat/test-stroke.R [hunk 2]
```diff
@@ -195,3 +195,32 @@ test_that("a ring is recognized when from_edge is specified", {
...
   expect_setequal(actual, expected)
 })
+
+test_that("flow mode does not break edges on a real dataset", {
+  edges <- sf::st_geometry(bucharest$streets)
+
+  strokes <- rcoins::stroke(edges, flow_mode = TRUE)
+
+  # find out which of the initial edges are contained in each of the strokes
+  # NOTE: edges included in self-interesecting strokes can be missed by the
```
Comment (cforgaci) [inline tests/testthat/test-stroke.R:205]:
```suggestion
  # NOTE: edges included in self-intersecting strokes can be missed by the
```

File: tests/testthat/test-stroke.R [hunk 3]
```diff
@@ -195,3 +195,32 @@ test_that("a ring is recognized when from_edge is specified", {
...
+    } else {
+      return(sf::st_line_merge(union))
+    }
+  }
+  edges_merged <- sf::st_sfc(sapply(contains, merge_edges),
+                             crs = sf::st_crs(edges))
+
+  # compare the grouped edges to the strokes: if identical, this means that
+  # the strokes contain full edges, i.e. flow_mode is respected
+  # NOTE: elemet-wise comparison works even if "strokes" consists of only
```
Comment (cforgaci) [inline tests/testthat/test-stroke.R:223]:
```suggestion
  # NOTE: element-wise comparison works even if "strokes" consists of only
```

**Review by cforgaci — APPROVED** [review]
👍 + a couple of typos. Thanks!
